aria_data_loaders/aria_data_utils/episodic_memory_robotic_fetch.py

```diff
@@ -0,0 +1,467 @@
+import logging
+from typing import Any, Dict, List, Optional, Tuple
+
+import click
+import numpy as np
+import rospy
+import sophus as sp
+from aria_data_utils.conversions import (
+    ros_PoseStamped_to_sp_SE3,
+    ros_TransformStamped_to_sophus_SE3,
+    sophus_SE3_to_ros_PoseStamped,
+    sophus_SE3_to_ros_TransformStamped,
+    xyt_to_sophus_SE3,
+)
+from geometry_msgs.msg import PoseStamped
+from spot_rl.envs.skill_manager import SpotSkillManager
+from spot_rl.utils.utils import ros_frames as rf
+from spot_wrapper.spot import Spot
+from spot_wrapper.spot_qr_detector import SpotQRDetector
+from std_msgs.msg import Bool
+from tf2_ros import (
+    ConnectivityException,
+    ExtrapolationException,
+    LookupException,
+    StaticTransformBroadcaster,
+)
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+# For information on useful reference frame, please check https://github.com/facebookresearch/spot-sim2real?tab=readme-ov-file#eyeglasses-run-spot-aria-project-code
+
+SPOT_DOCK_OBSERVER_WAYPOINT_LEFT = [
```

Comment:
Just added a comment about them.

Basically, instead of starting spot in a pose where it can see the QR code, we are now starting it on the dock itself.

At the start of the demo, robot gets up from the dock, moves to the `SPOT_DOCK_OBSERVER_WAYPOINT_LEFT`, sits down there and starts looking at the QR code to register the marker in the TF tree.

For this specific purpose, we have 2 waypoints `SPOT_DOCK_OBSERVER_WAYPOINT_LEFT` and `SPOT_DOCK_OBSERVER_WAYPOINT_RIGHT`. User can choose between the 2 based on his layout of space.